### PR TITLE
New gromacs instructions

### PR DESCRIPTION
**Pull request by RuiApostolo**

### CODE REVIEW

File: apps/GROMACS/ARCHER2_2023.4_cpe15.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+export CC=cc
+export CXX=CC
+export GMX_INSTALL=/work/y07/shared/apps/core/gromacs/2023.4
+
+cmake                                       \
+      -DGMX_MPI=ON                          \
+      -DGMX_OPENMP=ON                       \
+      -DGMX_GPU=OFF                         \
+      -DGMX_DOUBLE=OFF                      \
+      -DGMX_BUILD_OWN_FFTW=ON               \
```
Comment (al42and) [inline apps/GROMACS/ARCHER2_2023.4_cpe15.md:57]: This line would make GROMACS ignore earlier loaded `cray-fftw`. That's not incorrect (one can build own FFTW or use Cray's, either works), but first loading the module and then ignoring it seems a bit strange.

File: apps/GROMACS/ARCHER2_2023.4_cpe15.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+prefix to somewhere you have permission to write to.
+
+```bash
+export CC=cc
+export CXX=CC
+export GMX_INSTALL=/work/y07/shared/apps/core/gromacs/2023.4
+
+cmake                                       \
+      -DGMX_MPI=OFF                         \
+      -DGMX_OPENMP=OFF                      \
```
Comment (al42and) [inline apps/GROMACS/ARCHER2_2023.4_cpe15.md:120]: While none of the analysis tools use MPI, some can use OpenMP to speed things up, so it's not required nor optimal to disable OpenMP.

**Review by al42and — COMMENTED** [review]
Sorry for barging in, but I noticed a few minor quirks in your instructions. Nothing big, but could be worth fixing whenever the next revision is made.

File: apps/GROMACS/ARCHER2_2023.4_cpe15.md (hunk 1 above)
Comment (RuiApostolo) [reply apps/GROMACS/ARCHER2_2023.4_cpe15.md:57]: Hi Andrey,

Thank you for taking the time to comment. I've had a look in our module, and it turns out that even though I was building the GROMACS FFTW routines, the cray ones were being picked up:

```
~> module load gromacs/2024.2
~> ldd /work/y07/shared/apps/core/gromacs/2024.2/bin/gmx_mpi | grep fftw
        libfftw3f_omp.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3f_omp.so.mpi31.3 (0x00007f780946c000)
        libfftw3_omp.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3_omp.so.mpi31.3 (0x00007f7809265000)
        libfftw3f.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3f.so.mpi31.3 (0x00007f780023e000)
        libfftw3.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3.so.mpi31.3 (0x00007f77ffd37000)
```

I'll update the documentation, removing the GROMACS-FFTW routine compilation.

File: apps/GROMACS/ARCHER2_2023.4_cpe15.md (hunk 2 above)
Comment (RuiApostolo) [reply apps/GROMACS/ARCHER2_2023.4_cpe15.md:120]: Thanks for the explanation, I've modified the current module (2024.2) and newer versions will continue to use `-DGMX_OPENMP=ON` for the serial version.

**Pull request merged**